### PR TITLE
sdl: Make background color configurable

### LINKED ISSUE #364: My fear of red is stopping me from using Decaf
Problem: Red murdered my uncle. I can't open Decaf.

Solution: Add background color option to Decaf config.

### PR DESCRIPTION
**Pull request by VelocityRa**
```
"ui": {
    "background_color": {
        "red": 153,
        "green": 51,
        "blue": 51
    }
}
```
Resolves #364.

### CODE REVIEW

**Comment by VelocityRa** [comment]
Any problems I should address?

File: src/decaf-sdl/config.cpp [hunk 1]
```diff
@@ -322,7 +333,21 @@ save(const std::string &path)
...
 } // namespace config
+
+// External serialization functions
+namespace cereal
+{
+    // background_color_s serialization
+    template<class Archive>
+    void serialize(Archive &ar, decaf::config::ui::background_color_s &c)
+    {
+        ar(::cereal::make_nvp("red", c.r),
```
Comment (exjam) [inline src/decaf-sdl/config.cpp]: cereal::make_nvp, not ::cereal::make_nvp

File: src/decaf-sdl/decafsdl_opengl.h [hunk 2]
```diff
@@ -55,6 +55,8 @@ class DecafSDLOpenGL : public DecafSDLGraphics
    gl::GLuint mVertArray;
    gl::GLuint mVertBuffer;
    gl::GLuint mSampler;
+   
+   gl::GLfloat bg_col[3];
```
Comment (exjam) [inline src/decaf-sdl/decafsdl_opengl.h]: mBackgroundColour (or Color if you're a filthy american)

File: src/libdecaf/decaf_config.h [hunk 3]
```diff
@@ -96,6 +96,17 @@ extern double time_scale;
 
 } // namespace system
 
+namespace ui
+{
+
+extern struct background_color_s {
```
Comment (exjam) [inline src/libdecaf/decaf_config.h]: extern struct BackgroundColour

File: src/decaf-sdl/config.cpp (hunk 1 above)
Comment (VelocityRa) [inline src/decaf-sdl/config.cpp]: Whoops, refactored the function and forgot that there.

**Pull request merged**